examples/problem_decomposition/dmet.ipynb

```diff
@@ -804,6 +956,58 @@
     "vars(dmet_h4)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5.5 Restricting the fragment active space\n",
+    "\n",
+    "Problem decomposition is successful at reducing the problem size, but the subproblems still can demand too much resources for a quantum computer. It is possible to define an active space in each DMET fragment, and an example is shown below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{0: {'qubit_hamiltonian_terms': 27, 'circuit_width': 4, 'circuit_depth': 76, 'circuit_2qubit_gates': 46, 'circuit_var_gates': 12, 'vqe_variational_parameters': 2}, 1: {'qubit_hamiltonian_terms': 27, 'circuit_width': 4, 'circuit_depth': 76, 'circuit_2qubit_gates': 46, 'circuit_var_gates': 12, 'vqe_variational_parameters': 2}, 2: {'qubit_hamiltonian_terms': 361, 'circuit_width': 8, 'circuit_depth': 1573, 'circuit_2qubit_gates': 1180, 'circuit_var_gates': 160, 'vqe_variational_parameters': 14}, 3: {'qubit_hamiltonian_terms': 361, 'circuit_width': 8, 'circuit_depth': 1573, 'circuit_2qubit_gates': 1180, 'circuit_var_gates': 160, 'vqe_variational_parameters': 14}}\n"
+     ]
+    }
+   ],
+   "source": [
+    "def define_dmet_frag_as(homo_minus_m=0, lumo_plus_n=0):\n",
+    "    def callable_for_dmet_object(info_fragment):\n",
+    "        mf_fragment, _, _, _, _, _, _ = info_fragment\n",
+    "\n",
+    "        n_molecular_orb = len(mf_fragment.mo_occ)\n",
+    "\n",
+    "        n_lumo = mf_fragment.mo_occ.tolist().index(0.)\n",
+    "        n_homo = n_lumo - 1\n",
+    "\n",
+    "        frozen_orbitals = [n for n in range(n_molecular_orb) if n not in range(n_homo-homo_minus_m, n_lumo+lumo_plus_n+1)]\n",
+    "\n",
+    "        return frozen_orbitals\n",
+    "\n",
+    "    return callable_for_dmet_object\n",
+    "\n",
+    "options_h4_dmet = {\"molecule\": mol_h4,\n",
+    "                   \"fragment_atoms\": [1]*4,\n",
+    "                   # The fragment active space can be selected with a nested list on ints (corresponding \n",
+    "                   # to orbital indices), or a function.\n",
+    "                   \"fragment_frozen_orbitals\": [define_dmet_frag_as(0, 0), [0, 3, 4], define_dmet_frag_as(1, 1), None],\n",
+    "                   \"fragment_solvers\": \"vqe\",\n",
+    "                   \"solvers_options\": vqe_options, \n",
+    "                   \"verbose\": False}\n",
+    "\n",
+    "dmet_h4 = DMETProblemDecomposition(options_h4_dmet)\n",
+    "dmet_h4.build()\n",
+    "print(dmet_h4.get_resources())"
```

Comment:
Cool example !

Would mind adding pretty print (`pprint`) in the installs at the beginning and use pprint instead of print to help with readibility here ?